automerge/src/automerge.rs

```diff
@@ -182,6 +182,31 @@ impl Automerge {
     // PropAt::()
     // NthAt::()
 
+    pub fn parent_object<O: AsRef<ExId>>(&self, obj: O) -> Option<(ExId, crate::legacy::Key)> {
```

Comment:
I don't actually understand how this is compiling given that `crate::legacy::Key` is private :thinking: . In any case, I'm worried about exposing OpIDs as keys via `Key::ElementID`. Can we use indices instead? This would require a separate public `Key` enum I think.

Reply:
I think it is public but opaque. I've actually now updated it to use `Prop` since that seems to make more sense when returning the index. Note though that the index returned currently is the index of the op in the tree, not in the users list, I'm not sure how we could handle this.

Reply:
Seems like I could use `Nth` to convert the correct index to the one in the optree but I don't think we currently have a way of doing the inverse?